server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMailboxesMethodTest.java

```diff
@@ -778,8 +785,42 @@ public void getMailboxesShouldReturnUnlimitedQuotasForInboxByDefault() throws Ex
             .statusCode(200)
             .body(NAME, equalTo("mailboxes"))
             .body(ARGUMENTS + ".list", hasSize(1))
-            .body(FIRST_MAILBOX + ".quotas['key']['STORAGE'].max", nullValue())
-            .body(FIRST_MAILBOX + ".quotas['key']['MESSAGE'].max", nullValue());
+            .body(FIRST_MAILBOX + ".quotas['user-alice@domain.tld']['STORAGE'].max", nullValue())
+            .body(FIRST_MAILBOX + ".quotas['user-alice@domain.tld']['MESSAGE'].max", nullValue());
+    }
+
+    @Test
+    public void getMailboxesShouldReturnMaxStorageQuotasForInboxWhenSet() throws Exception {
+        quotaProbe.setDefaultMaxStorage(Optional.of(QuotaSize.size(42)));
+        String mailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, alice, DefaultMailboxes.INBOX).serialize();
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body("[[\"getMailboxes\", {\"ids\": [\"" + mailboxId + "\"]}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(ARGUMENTS + ".list", hasSize(1))
+            .body(FIRST_MAILBOX + ".quotas['user-alice@domain.tld']['STORAGE'].max", equalTo(42));
+    }
+
+    @Test
+    public void getMailboxesShouldReturnMaxMessageQuotasForInboxWhenSet() throws Exception {
+        quotaProbe.setDefaultMaxMessageCount(Optional.of(QuotaCount.count(43)));
```

Comment:
We should  introduce a version of `setDefaultMax\*` methods taking directly **long** as parameters....

Reply:
I don't think so, I like the Strong typing of these methods

Reply:
Ok, sorry it should take **QuotaCount** without the Optional to remove boiler plate stuff from logic.

Reply:
You are right. But matthieu changed again the API. It's worst with these SerializableValues, but I'm not sure we want to take the risk to enhance the probe API.